src/c++/library/http_client.cc

```diff
@@ -1466,10 +1466,15 @@ InferenceServerHttpClient::Infer(
   // RECV_END will be set.
   auto curl_status = curl_easy_perform(easy_handle_);
   if (curl_status == CURLE_OPERATION_TIMEDOUT) {
-    sync_request->http_code_ = 499;
+    std::cerr << "Curl failed with return code: " << curl_status << std::endl;
+    return Error(
+        "HTTP client failed (Deadline Exceeded): " +
```

Comment:
Note "Deadline Exceeded" is expected to be found in several tests, as well as being documented in the header, so I left it as part of the error message:
```
rmccormick@rmccormick-dt:~/triton/jira/4524-client-growth/server/qa$ grep -R -i deadline .
./L0_long_running_stress/scenarios.py:            assert False, "expected inference failure from deadline exceeded"
./L0_long_running_stress/scenarios.py:            if "Deadline Exceeded" not in ex.message():
./L0_client_timeout/client_timeout_test.py:        self.assertIn("Deadline Exceeded", str(cm.exception))
./L0_client_timeout/client_timeout_test.py:        self.assertIn("Deadline Exceeded", str(cm.exception))
./L0_client_timeout/client_timeout_test.py:        self.assertIn("Deadline Exceeded", str(cm.exception))
./L0_client_timeout/test.sh:if [ `grep -c "Deadline Exceeded" ${CLIENT_LOG}.c++.grpc_infer` != "1" ]; then
./L0_client_timeout/test.sh:if [ `grep -c "Deadline Exceeded" ${CLIENT_LOG}.c++.grpc_async_infer` != "1" ]; then
./L0_client_timeout/test.sh:if [ `grep -c "Deadline Exceeded" ${CLIENT_LOG}.c++.http_infer` == "0" ]; then
./L0_client_timeout/test.sh:if [ `grep -c "Deadline Exceeded" ${CLIENT_LOG}.c++.http_async_infer` == "0" ]; 
```